### ISSUE TITLE
VWAP indicator comparison to TA-Lib

### ISSUE THREAD

**Issue by esmeraldaliaj** [twopirllc/pandas-ta#275]
I am not sure if this is simply my mistake or a bug (I guess someone would have seen it by now), but I was testing the VWAP indicator and I got something like the following

![image](https://user-images.githubusercontent.com/5569899/115269032-c1900080-a0ef-11eb-8e35-9fe7be82026b.png)

the **red** line (this library) is the result of:

```
data["vwap"] = vwap(
        data["high"],
        data["low"],
        data["close"],
        data["volume"]
    )
```

The **golden** line (TA-Lib) is the result of:

```
data["vwap2"] = volume_weighted_average_price(
        high=data["high"],
        low=data["low"],
        volume=data["volume"],
        close=data["close"],
        window=20
    )
```

Using the following data:
`"ARKK", start_date="04/03/21", end_date="04/10/21", interval="1m"` from index _0_ to _385_

What might be the reason for this?

**Comment by twopirllc** [comment]
Hello @esmeraldaliaj,

Thanks for providing _some_ details **but** unfortunately it is not enough for me to reproduce and compare.

Also **TA Lib** (python wrapper) does not have [**volume_weighted_average_price**](https://github.com/mrjbq7/ta-lib/search?q=volume_weighted_average_price&type=) method nor does it exist in the **TA Lib** C Library which the Python Wrapper is based on. Is _volume_weighted_average_price_ a custom function of yours?

Additionally, [**VWAP**](https://www.investopedia.com/terms/v/vwap.asp) is **not** a rolling average and has **no** ```window``` parameter. So that is one reason why there is a _major_ difference in your chart.

Below is documentation of Pandas TA's **VWAP**. It is anchored by default to "D" (or "Daily") which was requested so it matches TradingView's Anchored VWAP. Since you are using "minute" data, you will need to change the anchor with the available types listed in the Pandas Library as noted in the **vwap** documentation below.

<img width="922" alt="Screen Shot 2021-04-19 at 1 31 24 PM" src="https://user-images.githubusercontent.com/18198392/115299651-e9915b00-a113-11eb-96e4-3064e6c393c0.png">

In summary, please provide **more** code as well as share your ```"ARKK" 1min``` data with me so I can be more helpful in clarifying the discrepancy.

Kind Regards,
KJ

**Comment by esmeraldaliaj** [comment]
@twopirllc you are absolutely correct, I meant to say I was comparing it to the function found here `https://github.com/bukosabino/ta` [`from ta.volume import volume_weighted_average_price`].

With this data
[arkk.csv](https://github.com/twopirllc/pandas-ta/files/6343941/arkk.csv)

This is nothing urgent by the way, just making sure that everything works fine. Otherwise great library.

**Comment by twopirllc** [comment]
Hello @esmeraldaliaj,

Apologies for taking so long to answering this.

The differences lie in that [Bukosabino's TA](https://github.com/bukosabino/ta/blob/84ced26e07cb676d1033703dfd81e1ee81285b07/ta/volume.py#L457) uses a _rolling_ **sum** for it's calculation while Trading View and Pandas TA utilize a **cumsum**. Bukosabino's is a custom version and not in line with the definition of _VWAP_ and that is ok if it suits your needs and helps you resolve your analysis.

<br/>

**Trading View's** and **Pandas TA's**
<img width="596" alt="Screen Shot 2021-06-21 at 10 35 28 AM" src="https://user-images.githubusercontent.com/18198392/122805099-817b0480-d27d-11eb-8399-711a11d8dd8f.png">

<br/>

I have included some code that compares Pandas TA's VWAP Daily and Hourly and VWMA(20) with Bukosabino's TA VWAP(20).
```python
import pandas as pd
import pandas_ta as ta
import ta as bta

def pdata(path="../arkk.csv", length=20, cols=["date", "open", "high", "low", "volume"]):
    df = pd.read_csv(path)
    df.drop(columns=["ticker"], errors="ignore", inplace=True)
    df.set_index(pd.DatetimeIndex(df["date"]), inplace=True)
    
    # Pandas TA
    df.ta.vwap(anchor="D", append=True) # Daily: red
    df.ta.vwap(anchor="H", append=True) # Hourly: blue
    
    # Pandas TA VWMA similar to Bukosabino TA VWAP: green
    df.ta.vwma(length=20, append=True)
    
    # Bukosabino TA: orange
    df[f"BTA_VWAP_{length}"] = bta.volume.volume_weighted_average_price(df.high, df.low, df.close, df.volume, length)
    
    return df.drop(cols, axis=1)

df = pdata(length=20)

title = f"ARKK [{df.index[0]} to {df.index[385]}]"
colors = ["black", "red", "blue", "green", "orange"]
df.iloc[:385].plot(figsize=(16,8), title=title, color=colors, grid=True)
```

<img width="864" alt="Screen Shot 2021-06-21 at 10 47 52 AM" src="https://user-images.githubusercontent.com/18198392/122805929-8d1afb00-d27e-11eb-8b47-6dad442dd1a8.png">

Hope this helps clarify the differences, but use what is relevant and useful for your use case.

Kind regards,
KJ

**Comment by twopirllc** [comment]
Hello @esmeraldaliaj,

I assume by no response that the solution provided was sufficient. Thus I will be closing this issue in a few days.

Kind Regards,
KJ

**Comment by jasongoodwin** [comment]
This looks fine to me. VWAP is anchored. Human traders that study vwap will usually use multiple anchor points (pivot points, 1 week, 1 month, daily or session)
